tools/pylib/_boutpp_build/backend.py

```diff
@@ -0,0 +1,216 @@
+import os  # corelib
+import glob  # corelib
+import hashlib  # corelib
+import base64  # corelib
+import tempfile  # corelib
+import subprocess  # corelib
+
+try:
+    import packaging.tags  # packaging
+except:
+    packaging = None
+
+
+def run(cmd):
+    print(f"running `{cmd}`")
+    ret = os.system(cmd)
+    assert ret == 0, f"{cmd} failed with {ret}"
+
+
+useLocalVersion = True
+
+
+def getversion(_cache={}):
+    if "r" not in _cache:
+        try:
+            tmp = run2("git describe --tags --match=v4.0.0")
+            if useLocalVersion:
+                tmp = run2(
+                    f"echo {tmp.strip()} | sed -e s/v4.0.0-/v5.0.0.dev/ -e s/-/+/"
+                )
+            else:
+                tmp = run2(
+                    f"echo {tmp.strip()} | sed -e s/v4.0.0-/v5.0.0.dev/ -e s/-.*//"
+                )
```

Comment:
Why not `re.sub` rather than starting another process?

Reply:
Copy-pased from the cmake version. But yes, `re.sub` is the way to go.